server/controllers/userController.js

```diff
@@ -0,0 +1,163 @@
+   * @returns {object} Object containing token to the user
+   * @memberof UsersController
+   */
+  static async login(req, res, next) {
+    try {
+      const data = req.body;
+      const { userLogin, password } = data;
+      // Login with username or email address
+      const user = await userModel.findUserInput(userLogin);
+      if (!user) {
+        return res.status(404).json({
+          status: 'fail',
+          data: { message: 'Invalid Login Details' }
+        });
+      }
+      const passwordValid = await decryptPassword(password, user.password);
+      if (!passwordValid) {
+        return res.status(401).json({
+          status: 'fail',
+          data: { message: 'Invalid Login Details' }
+        });
+      }
+      const token = await generateToken(user);
+      return res.status(200).json({
+        status: 'success',
+        data: { token }
+      });
+    } catch (err) {
+      return next(err);
+    }
+  }
+
+  /**
+   *
+   * Forgot Password Middleware - Enables users request new password
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns {object} Object containing token to the user
+   * @memberof UsersController
+   */
+  static async forgotPassword(req, res, next) {
+    try {
+      const { userLogin } = req.body;
+      const user = await userModel.findUserInput(userLogin);
+      if (!user) {
+        return res.status(404).json({
+          status: 'fail',
+          data: { message: 'Invalid Login Details' }
+        });
+      }
+      const token = await crypto.randomBytes(20).toString('hex');
+      const mailSubject = 'Reset your password';
+      const mailContent = `You are receiving this mail because you requested
+       for a password change. Please Click on the link below or paste it
+       in a browser to complete the process <br />
+        http://${req.headers.host}/reset/${token} <br/>
+        Link expires in 1 hour time. <br />
+       If you did not request this, please Ignore this email and your
+       password will remain unchanged.
+      `;
+      await emailModule.sendEmailToUser(userLogin, mailSubject, mailContent);
+      await userModel.updateToken(token, userLogin);
+      await userModel.updatePasswordExpires(Date.now() + 3600000, userLogin);
+      return res.status(200).json({
+        status: 'success',
+        data: { token }
+      });
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  /**
+   *
+   * Reset Password Middleware - Enables users reset their password
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns {object} Object containing token to the user
+   * @memberof UsersController
+   */
+  static async resetPassword(req, res, next) {
+    if (req.body.newPassword !== req.body.confirmPassword) {
+      return res.status(400).json({
+        status: 'fail',
+        data: { message: 'Password does not match' }
```

Comment:
the error message should be clearer,  e.g  `confirmPassword` and `password` are not equal

Reply:
Okay I'll ensure I edit the error message